azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -134,9 +135,15 @@ def update(self):
         incarnation, xml_text, xml_doc = GoalState._fetch_goal_state(self._wire_client)
         goal_state_updated = incarnation != self._incarnation
         if goal_state_updated:
-            logger.info('Fetched new goal state from the WireServer [incarnation {0}]', incarnation)
+            logger.info('Fetched a new incarnation for the WireServer goal state [incarnation {0}]', incarnation)
+
+        vm_settings, vm_settings_updated = None, False
+        try:
+            vm_settings, vm_settings_updated = GoalState._fetch_vm_settings(self._wire_client)
+        except VmSettingsSupportStopped as exception:  # If the HGAP stopped supporting vmSettings, we need to use the goal state from the WireServer
+            self._restore_wire_server_goal_state(incarnation, xml_text, xml_doc, exception)
+            return
```

Comment:
This handles  the exception that the HostPluginProtocol raises when support for vmSettings stops.